Sources/Receipt/ReceiptDetailSectionData.swift

```diff
@@ -0,0 +1,121 @@
+            type: receipt.type.rawValue.lowercased().localized(),
+            entry: receipt.entry.rawValue,
+            amount: receipt.amount,
+            currency: receipt.currency,
+            createdOn: ISO8601DateFormatter
+                .ignoreTimeZone
+                .date(from: receipt.createdOn)!
+                .format(for: .date),
+            iconFont: HyperwalletIcon.of(receipt.entry.rawValue).rawValue)
+    }
+}
+
+struct ReceiptDetailSectionDetailData: ReceiptDetailSectionData {
+    var rows = [(title: String, value: String)] ()
+    var receiptDetailSectionHeader: ReceiptDetailSectionHeader { return .details }
+    var rowCount: Int { return rows.count }
+    var cellIdentifier: String { return ReceiptDetailTableViewCell.reuseIdentifier }
+
+    init(from receipt: HyperwalletReceipt) {
+        let receiptId = receipt.journalId
+        rows.append((title: "receipt_details_receipt_id".localized(), value: receiptId))
+
+        let dateTime = ISO8601DateFormatter.ignoreTimeZone.date(from: receipt.createdOn)!.format(for: .dateTime)
+        rows.append((title: "receipt_details_date".localized(), value: dateTime))
+
+        if let charityName = receipt.details?.charityName {
+            rows.append((title: "receipt_details_charity_name".localized(), value: charityName))
+        }
+        if let checkNumber = receipt.details?.checkNumber {
+            rows.append((title: "receipt_details_check_number".localized(), value: checkNumber))
+        }
+        if let clientPaymentId = receipt.details?.clientPaymentId {
+            rows.append((title: "receipt_details_client_payment_id".localized(), value: clientPaymentId))
+        }
+        if let website = receipt.details?.website {
+            rows.append((title: "receipt_details_website".localized(), value: website))
+        }
+    }
+}
+
+struct ReceiptDetailSectionFeeData: ReceiptDetailSectionData {
+    var rows = [(title: String, value: String)]()
+    var receiptDetailSectionHeader: ReceiptDetailSectionHeader { return .fee }
+    var rowCount: Int { return rows.count }
+    var cellIdentifier: String { return ReceiptFeeTableViewCell.reuseIdentifier }
+
+    init(from receipt: HyperwalletReceipt) {
+        let amountFormat = receipt.entry == HyperwalletReceipt.HyperwalletEntryType.credit ? "+%@ %@" : "-%@ %@"
+        let valueCurrencyFormat = "%@ %@"
+        rows.append(
+            (title: "receipt_details_amount".localized(),
+             value: String(format: amountFormat, receipt.amount, receipt.currency)))
+        var fee: Float = 0.0
+        if let strFee = receipt.fee {
+            rows.append((title: "receipt_details_fee".localized(),
+                         value: String(format: valueCurrencyFormat, strFee, receipt.currency)))
+            fee = Float(strFee) ?? 0.0
+        }
+        if let amount = Float(receipt.amount) {
+            let transaction = String(format: "%.2f", amount - fee)
```

Comment:
If entry = debit, transaction should be -amount - fee and for credit it should be amount - fee

Reply:
fixed